src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -3096,7 +3102,11 @@ PartitionColumnOpExpressionList(Query *query)
 		rangeTableEntryIndex = candidatePartitionColumn->varno - 1;
 		rangeTableEntry = list_nth(rangetableList, rangeTableEntryIndex);
 
-		Assert(rangeTableEntry->rtekind == RTE_RELATION);
+		/* when user refers to a column from the JOIN instead of the relation */
```

Comment:
I can see that the user did, but why can we skip it? :)

Reply:
Good point 😄  

Clarified the comment. But basically its not very trivial to implement and the code path is obsolete that we should remove (see #1204 sub-item: `Filters on the LEAF queries`).

So, I didn't find it very useful to fix that for now.